.github/workflows/drepecated-functions-checker.yml

```diff
@@ -0,0 +1,67 @@
+name: "Check Deprecations"
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled]
+    # Runs on PRs to main
+    branches:
+      - main
+
+jobs:
+  deprecations-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0 # by default the checkout action doesn't checkout all branches
+      - name: Set up Go
+        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 #v4
+        with:
+          go-version-file: ./.go-version
+          cache: true
+      - name: Check deprecations for files in diff
+        run: |  
+          # Staticcheck uses static analysis to finds bugs and performance issues, offers simplifications, 
+          # and enforces style rules.
+          # Here, it is used to check if a deprecated function, variable, constant or field is used.
+
+          echo "Installing staticcheck"
+          go install honnef.co/go/tools/cmd/staticcheck@2023.1.2 #v0.4.2
+
+          # Run staticcheck 
+          echo "Performing Deprecations Check: Running Staticcheck"
+          staticcheck ./... | grep deprecated > staticcheckOutput.txt
+
+          # Get changed files names from the PR
+          changedFiles=$(git --no-pager diff --name-only HEAD "$(git merge-base HEAD "origin/${{ github.event.pull_request.base.ref }}")")
```

Comment:
did you consider using [revgrep](https://pkg.go.dev/github.com/golangci/revgrep#section-readme) to narrow down this to only new additions that have deprecated functions? The advantage of revgrep is that the PR author doesn't need to fix the whole file for the CI check to go green, they only need to fix any new code that they're writing.

Reply:
This is a great tool !! I was looking something to do this and most of the commands I found online did not work and I gave up. I made changes using revgrep now. Thanks